### PR TITLE
expression: fix the incorrect result of `json_length(cast(1 as json), null)`

### LINKED ISSUE #50931: The result of `json_length(cast(1 as json), null)` isn't same as mysql
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
```
select json_length(1, null);
```

### 2. What did you expect to see? (Required)
```
+------------------------------------+
| json_length(cast(1 as json), null) |
+------------------------------------+
|                               NULL |
+------------------------------------+
```

### 3. What did you see instead (Required)
```
+------------------------------------+
| json_length(cast(1 as json), null) |
+------------------------------------+
|                                  1 |
+------------------------------------+
```

### 4. What is your TiDB version? (Required)

Git Commit Hash: c0476a0e8ca2cefd50a7cccf3a7e9d2deeb5fee3

### PR DESCRIPTION
**Pull request by SeaRise**
### What problem does this PR solve?

Issue Number: close #50931

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/expression/builtin_json_vec.go [hunk 1]
```diff
@@ -860,6 +851,11 @@ func (b *builtinJSONLengthSig) vecEvalInt(ctx EvalContext, input *chunk.Chunk, r
 				return types.ErrInvalidJSONPathMultipleSelection
 			}
 
+			jsonItem := jsonBuf.GetJSON(i)
+			if jsonItem.TypeCode != types.JSONTypeCodeObject && jsonItem.TypeCode != types.JSONTypeCodeArray {
```
Comment (windtalker) [inline pkg/expression/builtin_json_vec.go:855]: In non-vector version, if `path` does not exists, it returns null, but in the vector version, if the jsonItem is not JSON object or JSON array, it returns 1 even if `path` does not exists? According to MySQL's behavior, the non-vector version is right.

Comment (SeaRise) [reply pkg/expression/builtin_json_vec.go:855]: `result.MergeNulls(pathBuf)` has been call in L839, so if `path` is null, `continue` will be call in L843

Comment (windtalker) [reply pkg/expression/builtin_json_vec.go:855]: I mean the `path` is not null, but a non exists path. Like `select json_length(cast(1 as json), '$.a')`

Comment (SeaRise) [reply pkg/expression/builtin_json_vec.go]: get! fixed and added test case.

**Review by YangKeao — APPROVED** [review]
LGTM

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**